diffhtwo/experimental/tests/test_diffstarpop_halpha.py

```diff
@@ -0,0 +1,133 @@
+import numpy as np
+from diffstar.defaults import T_TABLE_MIN
+
+from diffstarpop.defaults import DEFAULT_DIFFSTARPOP_PARAMS
+
+from dsps.cosmology import flat_wcdm
+from dsps.cosmology.defaults import DEFAULT_COSMOLOGY
+from dsps.metallicity import umzr
+from dsps.data_loaders import retrieve_fake_fsps_data
+
+
+from diffsky.experimental import mc_lightcone_halos as mclh
+from .. import diffstarpop_halpha
+from diffsky.param_utils import spspop_param_utils as spspu
+
+# from diffsky.experimental.scatter import DEFAULT_SCATTER_PARAMS
+# from diffsky.ssp_err_model import ssp_err_model
+
+import jax.numpy as jnp
+from jax import random as jran
+from jax import jit as jjit
+
+
+@jjit
+def diffstarpop_halpha_kern():
+    ssp_data = retrieve_fake_fsps_data.load_fake_ssp_data()
+
+    # get fake ssp_halpha_luminosity
+    arr = jnp.array(
+        [
+            18.99350973,
+            15.18340321,
+            18.59590888,
+            14.51430117,
+            15.50420145,
+            18.89018259,
+            14.6840321,
+            15.44170473,
+            18.62229038,
+            18.23519039,
+            15.46710684,
+            18.09569229,
+            18.72593381,
+            18.74917029,
+            18.84602614,
+            18.90889461,
+            20.18508488,
+            19.64529091,
+            20.39642718,
+            19.94077289,
+            20.36111634,
+            20.77151474,
+            21.80102465,
+            22.86217362,
+            24.00485424,
+            25.53092415,
+            26.12223314,
+            29.38864892,
+            30.45950998,
+            34.16892198,
+            39.6796686,
+            39.59340563,
+            33.72510107,
+            29.60391916,
+            23.58247219,
+            16.62595202,
+            13.70698426,
+            8.85376548,
+            2.83265497,
+            2.0991581,
+            1.48929108,
+            0.96965843,
+            0.58217645,
+            0.35032972,
+            0.22578957,
+            0.14377021,
+            0.09140591,
+        ]
+    )
+    noise_std = 2
+    ssp_halpha_luminosity = [
+        np.clip(
+            arr + np.random.normal(0, noise_std, size=arr.shape), a_min=0, a_max=None
```

Comment:
In unit tests where there is stochasticity, I try to use `jax.random` rather than `np.random`, because then the test has fully deterministic behavior. Otherwise, a unit test can sometimes pass, and other times fail, depending the random seed used in the testing, which can create confusion. But this is a very minor thing and not worth changing here.